Makefile

```diff
@@ -35,8 +46,24 @@ PKG:=k8s.io/client-go/pkg
 VERSION_LDFLAGS:=-X $(PKG)/version.gitVersion=$(GIT_TAG) -X $(PKG)/version.gitCommit=$(GIT_COMMIT) -X $(PKG)/version.buildDate=$(BUILD_DATE)
 LDFLAGS:=-w $(VERSION_LDFLAGS)
 
-metrics-server: $(SRC_DEPS)
-	GOARCH=$(ARCH) CGO_ENABLED=0 go build -mod=readonly -ldflags "$(LDFLAGS)" -o metrics-server sigs.k8s.io/metrics-server/cmd/metrics-server
+metrics-server:
+	OUTPUT_DIR=. BINARY_NAME=$@ $(MAKE) binary
+
+.PHONY: binary
+binary: $(SRC_DEPS)
+	@mkdir -p $(OUTPUT_DIR)
+	GOARCH=$(ARCH) GOOS=$(OS) CGO_ENABLED=0 go build -mod=readonly -ldflags "$(LDFLAGS)" -o "$(OUTPUT_DIR)/$(BINARY_NAME)" sigs.k8s.io/metrics-server/cmd/metrics-server
+
+.PHONY: binary-all
+binary-all:
+	@for arch in $(ALL_ARCHITECTURES); do \
+		for platform in $(ALL_PLATFORMS); do \
+		    if ! echo "$(ALL_GO_DIST)" | grep -q "$${platform}/$${arch}" ; then \
```

Comment:
Not sure if I like the fact that list of binaries published depends on local golang. Would prefer to have hardcoded list and have release fail if local go doesn't support this

Reply:
I do not have a strong opinion.